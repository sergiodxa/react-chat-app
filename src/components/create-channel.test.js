import React from "react";
import { render, fireEvent, waitForElement } from "@testing-library/react";

import CreateChannel from "./create-channel";

test("CreateChannel", async () => {
  const onCreate = jest.fn();
  const channels = ["general", "testing"];
  const { getByLabelText } = render(
    <CreateChannel channels={channels} onCreate={onCreate} />
  );

  const button = getByLabelText("Open create channel dialog");
  fireEvent.click(button);

  const input = await waitForElement(() => getByLabelText("Channel name"));
  fireEvent.change(input, {
    target: {
      value: "react"
    }
  });
  fireEvent.submit(input);

  expect(onCreate).toHaveBeenCalledWith("react");
});

test("CreateChannel UI", async () => {
  const { getByLabelText, asFragment } = render(
    <CreateChannel channels={["general"]} onCreate={jest.fn()} />
  );

  expect(asFragment()).toMatchSnapshot();

  fireEvent.click(getByLabelText("Open create channel dialog"));
  await waitForElement(() => getByLabelText("Channel name"));
  expect(asFragment()).toMatchSnapshot();

  fireEvent.click(getByLabelText("Close create channel dialog"));
  expect(asFragment()).toMatchSnapshot();
});
