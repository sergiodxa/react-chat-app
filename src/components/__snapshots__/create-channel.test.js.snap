// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateChannel UI 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: none;
  border: none;
  color: black;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  outline: none;
  padding: 1rem .5rem;
  width: 100%;
}

<button
    aria-label="Open create channel dialog"
    class="emotion-0"
    type="button"
  >
    Create Channel 
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`CreateChannel UI 2`] = `
<DocumentFragment>
  .emotion-0 {
  background: none;
  border: none;
  color: black;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  outline: none;
  padding: 1rem .5rem;
  width: 100%;
}

<button
    aria-label="Open create channel dialog"
    class="emotion-0"
    type="button"
  >
    Create Channel 
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`CreateChannel UI 3`] = `
<DocumentFragment>
  .emotion-0 {
  background: none;
  border: none;
  color: black;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  outline: none;
  padding: 1rem .5rem;
  width: 100%;
}

<button
    aria-label="Open create channel dialog"
    class="emotion-0"
    type="button"
  >
    Create Channel 
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
      />
    </svg>
  </button>
</DocumentFragment>
`;
